### PR TITLE
feat: [FFM-6667]: flag to separate MFE targets detail page

### PR DESCRIPTION
**Pull request by crystal-fong**
### Summary
The epic in reference https://harness.atlassian.net/browse/FFM-6665

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/RouteDestinations.tsx [hunk 1]
```diff
@@ -249,9 +250,19 @@ const CFRoutes: FC = () => {
...
       >
-        <FFGitSyncProvider> {FFM_1827 ? <TargetDetailPage /> : <LegacyTargetDetailPage />} </FFGitSyncProvider>
+        <FFGitSyncProvider>
+          {FFM_1827 && !FFM_6665_FF_MFE_Target_Detail ? (
+            <TargetDetailPage />
+          ) : !FFM_1827 && FFM_6665_FF_MFE_Target_Detail ? (
+            <FFUIApp />
+          ) : (
+            <LegacyTargetDetailPage />
+          )}
```
Comment (knagurski) [inline src/modules/75-cf/RouteDestinations.tsx:264]: We can assume that the MFE will take precedence here, regardless of other flags. As-is, if we turn on the MFE for an env where the newer page is enabled (which is everywhere, for all customers, except with us in UAT), they'll actually get the legacy version. Maybe try something like this...
```suggestion
          {FFM_6665_FF_MFE_Target_Detail
            ? <FFUIApp />
            : FFM_1827 ? <TargetDetailPage /> : <LegacyTargetDetailPage />
          }
```

Comment (crystal-fong) [reply src/modules/75-cf/RouteDestinations.tsx]: fair enough, updated now 👍

**Review by laura-mannus — APPROVED** [review]
Good stuff :)

**Review by knagurski — APPROVED** [review]
I hereby approve and stuff

**Pull request merged**